vault/auth.go

```diff
@@ -820,6 +824,23 @@ func (c *Core) setupCredentials(ctx context.Context) error {
 			}
 		}
 
+		// Do not start up deprecated builtin plugins. If this is a major/minor
+		// upgrade, stop unsealing and shutdown. If we've already mounted this
+		// plugin, skip backend initialization and mount the data for posterity.
+		if versions.IsBuiltinVersion(entry.RunningVersion) {
+			shutdown := isMajorOrMinorUpgrade(c.currentVaultVersion.Version, entry.LastMounted)
+			_, err := c.handleDeprecatedMountEntry(ctx, entry, consts.PluginTypeCredential)
+			if shutdown && err != nil {
+				go c.ShutdownCoreError(err)
+				return errLoadAuthFailed
+			} else if err != nil {
+				backend.Cleanup(ctx)
+				backend = nil
+				c.logger.Error("skipping deprecated credential entry", "path", entry.Path, "error", err)
+				goto ROUTER_MOUNT
```

Comment:
If we're skipping it why should we goto ROUTER_MOUNT?

Reply:
This mounts the backend and skips initialization (see logic for non-builtin skipping above).

Reply:
I'm sorry, I'm not following.

Reply:
My understanding is this logic is a corollary to the `isMountable` [logic](https://github.com/hashicorp/vault/blob/main/vault/auth.go#L800-L806). `newCredentialBackend` is guaranteed to return a `nil` backend, which is not true here, so to match that behavior we just clean up the backend, set it to `nil`, and continue mounting.